### ISSUE TITLE
We want to use a fully supervised approach to train a model on the robotcar dataset?

### ISSUE THREAD

**Issue by RuiTianHIT** [md4all/md4all#35]
@sgasperini @morbi25 
Hello author!
md4all is a good work, we have been following and following this paper.
There is a problem now.
We want to train the model in the robotcar dataset with a fully supervised paradigm. We simply modified your code, but an error occurred.
![1](https://github.com/md4all/md4all/assets/104761701/455fc2c1-1328-4dff-9620-0a3b3066575c)
![2](https://github.com/md4all/md4all/assets/104761701/f48065c4-df8e-4380-bfdf-65a4a8fe6f76)

Is it possible that robotcar cannot access the depth gt of the training set, and therefore cannot train the model in the robotcar dataset with a fully supervised paradigm?

Looking forward to the author's reply, thank you very much!

**Comment by sgasperini** [comment]
Hi @RuiTianHIT,

We do not make use of the gt for the training set on robotcar, as we used a self-supervised method there, so we did not prepare the training dataset for that.

I believe you have to first prepare the ground truth data for the training set to be in the expected format. We did that only for the test set. Following our readme and adapting it to the training scenes should get you there.

Thanks

**Comment by RuiTianHIT** [comment]
@sgasperini 
Thank you for the author's reply.
We noticed the command to get the depth gt in readme.
![3](https://github.com/md4all/md4all/assets/104761701/6311e0da-02d6-48db-bcf8-fc21648ac7e4)
Can we get the depth gt of trainset by making the following modifications?
\ie python data/robotcar/precompute_depth_gt.py --dataroot <PATH_TO_DATAROOT> --scenes 2014-12-09-13-21-02 2014-12-16-18-44-24 --mode train val test

**Comment by sgasperini** [comment]
Hi @RuiTianHIT,

You will need to select the specific training scenes. Those are the test scenes. And I don't know what --mode train does, it seems it changes the timestamps. In theory also val test produces the gt. That script is from robotcar. You can try to launch it as a separate command just with --mode train on a different scene and then check if what comes out is what you want.

**Comment by morbi25** [comment]
Hi! Closing this for inactivity. Feel free to reopen it if you have follow-up questions.
